libr/anal/meta.c

```diff
@@ -540,10 +540,12 @@ R_API int r_meta_list_cb(RAnal *a, int type, int rad, SdbForeachCallback cb, voi
 	}
 	isFirst = true; // TODO: kill global
 	ls_foreach (ls, lsi, kv) {
-		if (cb) {
-			cb ((void *)&ui, kv->key, kv->value);
-		} else {
-			meta_print_item ((void *)&ui, kv->key, kv->value);
+		if (type == R_META_TYPE_ANY || (strlen (kv->key) > 5 && kv->key[5] == type)) {
```

Comment:
why kv->key[5] :?

Reply:
That's because of [this line in meta_print_item](https://github.com/radare/radare2/blob/86a3aa92b6792e6c6a18311317eddd5eb492ce21/libr/anal/meta.c#L500). I figured that there's no need to repeat the checks before since if `key[5]` is not a type then printing will be aborted either in `meta_print_item` or (by accident) in `r_meta_list_cb`.